### PR TITLE
Fixes for doing inference with masked padding

### PR DESCRIPTION
**Pull request by coryMosaicML**
A few inference fixes:

- Pass keyword arguments to inference models
- Add option to disable xformers
- Fix a bug  in constructing masks in the generate function

### CODE REVIEW

File: diffusion/models/stable_diffusion.py [hunk 1]
```diff
@@ -473,12 +473,13 @@ def generate(
...
                 pooled_embeddings = torch.cat([pooled_unconditional_embeddings, pooled_text_embeddings])  # type: ignore
+            if pad_attn_mask is not None:
+                encoder_attn_mask = torch.cat([uncond_pad_attn_mask, pad_attn_mask])  # type: ignore
         else:
             if self.sdxl:
                 pooled_embeddings = pooled_text_embeddings
-
             if pad_attn_mask is not None:
-                encoder_attn_mask = torch.cat([uncond_pad_attn_mask, pad_attn_mask])  # type: ignore
+                encoder_attn_mask = pad_attn_mask
```
Comment (jazcollins) [inline diffusion/models/stable_diffusion.py:482]: This part shouldn't be necessary since its done on L458 (https://github.com/mosaicml/diffusion/blob/effe5145ec138d14f82f049b1024af015aaa047a/diffusion/models/stable_diffusion.py#L458C9-L458C85)

Sorry about the assigning two variables in one line, it is not the most readable. Prob would be better to break up L458 into:

```
pooled_embeddings = pooled_text_embeddings
encoder_attn_mask = pad_attn_mask
```

Comment (coryMosaicML) [reply diffusion/models/stable_diffusion.py]: Should be fixed now!
